tap_snowflake/client.py

```diff
@@ -3,67 +3,228 @@
+            "password": config["password"],
+        }
 
-    @staticmethod
-    def to_jsonschema_type(sql_type: sqlalchemy.types.TypeEngine) -> dict:
-        """Returns a JSON Schema equivalent for the given SQL type.
+        for option in ["database", "schema", "warehouse", "role"]:
+            if config.get(option):
+                params[option] = config.get(option)
 
-        Developers may optionally add custom logic before calling the default
-        implementation inherited from the base class.
-        """
-        # Optionally, add custom logic before calling the super().
-        # You may delete this method if overrides are not needed.
-        return super().to_jsonschema_type(sql_type)
+        return URL(**params)
 
-    @staticmethod
-    def to_sql_type(jsonschema_type: dict) -> sqlalchemy.types.TypeEngine:
-        """Returns a JSON Schema equivalent for the given SQL type.
 
-        Developers may optionally add custom logic before calling the default
-        implementation inherited from the base class.
-        """
-        # Optionally, add custom logic before calling the super().
-        # You may delete this method if overrides are not needed.
-        return super().to_sql_type(jsonschema_type)
+class SelectableSQLStream(SQLStream):
+    """Patches the parent classes get_records method to support stream property selection."""
 
+    def get_selected_columns(self, table) -> List:
+        """Filter column list according to selection criteria."""
+        return [
+            col.name for col in table.columns if self.mask[("properties", col.name)]
+        ]
 
-class SnowflakeStream(SQLStream):
+    def get_records(self, context: dict | None) -> Iterable[dict[str, Any]]:
+        """Return a generator of record-type dictionary objects.
+
+        If the stream has a replication_key value defined, records will be sorted by the
+        incremental key. If the stream also has an available starting bookmark, the
+        records will be filtered for values greater than or equal to the bookmark value.
+
+        Args:
+            context: If partition context is provided, will read specifically from this
+                data slice.
+
+        Yields:
+            One dict per record.
+
+        Raises:
+            NotImplementedError: If partition is passed in context and the stream does
+                not support partitioning.
+        """
+        if context:
+            raise NotImplementedError(
+                f"Stream '{self.name}' does not support partitioning."
+            )
+
+        table = self.connector.get_table(self.fully_qualified_name)
+        query = table.select()
+        selected_columns = self.get_selected_columns(table=table)
+        query = query.options(load_only(*selected_columns))
+        if self.replication_key:
+            replication_key_col = table.columns[self.replication_key]
+            query = query.order_by(replication_key_col)
+            start_val = (
```

Comment:
@aaronsteers we talked briefly about this; streamlining bookmark retrieval so the user doesn't have to check the bookmark type if they don't care about it (as in this case, where SQLAlchemy will handle type binding). You suggested `get_starting_replication_key_value` should suffice on its own, but the [doctoring](https://github.com/meltano/sdk/blob/861dfc327aadfc4f57557b81a95063d7e6628b1c/singer_sdk/streams/core.py#L214) suggests otherwise 🤔

> Developers should use this method to seed incremental processing for
        non-datetime replication keys. For datetime and date replication keys, use
        :meth:`~singer_sdk.Stream.get_starting_timestamp()`

Reply:
@kgpayne Passing a date-formatted string should suffice for most warehouses. They're smart enough to cast the literal into the right type:

```sql
select *
from my_table
where my_col >= '2022-01-01' -- the db casts this literal to the type of my_col
```

The docstring makes more sense for non-SQL taps.

Reply:
Yeah, I think either works. In the case of SQL taps and targets, the `Stream` class will not know (or care) what the type of the replication key is, so the generic makes sense.

Honestly, I am increasingly thinking we should just send the min/max constraints to the developer in the method arguments - so there's nothing to run and nothing to inspect aside from getting the value of `self.replication_key` to find which column name should be filtered in the SELECT query.

What do you guys think?

Related to my comment here: https://github.com/MeltanoLabs/tap-snowflake/pull/2#discussion_r982674005

Reply:
@aaronsteers I like the proposal. Left a comment in the discussion https://github.com/meltano/sdk/discussions/1011